server/services/link/MagicLinkService.ts

```diff
@@ -13,7 +13,7 @@ export default class MagicLinkService {
   async requestLink(email: string): Promise<unknown> {
     const magicLinkRequest: MagicLinkRequest = { email }
     return this.hmppsAuthClient.getSystemClientToken().then(hmppsToken =>
-      MagicLinkService.restClient(hmppsToken).post({
+      MagicLinkService.restClient(hmppsToken).update({
```

Comment:
Likewise - this is not an update operation.
I can now see why you were looking to explore other ways of refactoring the `restClient` class. As mentioned in my comment on it, I think the public API of the rest client should be rest/http terminology, because otherwise we end up with confusion like this at the service layer (ie. I've read this line and am immediately assuming I'm updating an existing magic link entity/resource, which is not the case)